### PR TITLE
make url validator more robust - fix issue #3175

### PR DESCRIPTION
**Pull request by hyi**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. The URL example shown in issue #3175 should be treated as a valid URL with this fix.

### CODE REVIEW

File: hs_core/views/utils.py [hunk 1]
```diff
@@ -124,7 +124,11 @@ def validate_url(url):
 
     # validate url is valid, i.e., can be opened
     try:
-        urlopen(url)
+        # have to add a User-Agent header and pass in a Request to urlopen to test 
+        # whether a url can be resolved since some valid website URLs block web 
+        # spiders/bots, which raises a 403 Forbidden HTTPError
+        url_req = Request(url, headers={'User-Agent': 'Mozilla/5.0'})
```
Comment (zhiyuli) [inline hs_core/views/utils.py]: Would this version number "5.0" get deprecated and cause issues as time goes?

Comment (hyi) [reply hs_core/views/utils.py]: @zhiyuli I have the same concern, but setting this User-Agent is the only one I can get from googling. If you can find a better one, please let me know.

**Comment by hyi** [comment]
test this, please

**Comment by hyi** [comment]
@zhiyuli can you approve this PR again so I can merge it now that all checks have passed?

**Comment by zhiyuli** [comment]
approved

**Pull request merged**